SU-HSDO/suhumsci
# SHS-5641: Site editors shouldn't remove imported photos

# READY FOR REVIEW

## Summary
Removes the ability for non-administrators from deleting media that is attached to nodes that have been imported via a migration
* Removes the "Delete Media item" contextual menu item in the node edit form
* Removes the "Remove" button on the media edit form
* Only applies to non-adminstrators
* Only applies to media that was part of a migration
* Note: Work was completed in a forked version of the `stanford_migrate` module. Forked version is located here: https://github.com/fourkitchens/stanford_migrate
* Note: There is a [PR over on the forked repo](https://github.com/fourkitchens/stanford_migrate/pull/2). Once that is merged in to the `8.x` branch over there, the `composer.json` file here will need to be modified to point to the main branch over there.

## Need Review By (Date)
2024-05-31

## Urgency
Medium

## Steps to Test
1. Login to a site as a non-admin
2. Make sure the site contains migrated nodes with media entities attached. Development and testing was done with the Physics site.
3. Edit a node that has been imported via a migration. Content types used for development and testing are Person and Event. You can tell if the node was part of a migration because there are some fields that are disabled and have an orange background:

![Screenshot 2024-05-30 at 3 55 31 PM](https://github.com/SU-HSDO/suhumsci/assets/107938318/77e425b1-2836-4355-bdba-1ae6bbe9631c)

4. Click on the contextual menu for the Image or Profile Image field
5. Verify that the "Delete Media item" is no longer available

![Screenshot 2024-05-30 at 3 56 37 PM](https://github.com/SU-HSDO/suhumsci/assets/107938318/df060d97-7099-42d8-ae00-c119e7926822)

6. Click "Edit media item"
7. Verify that on the media edit form, there is new messaging at the top informing that the media can not be deleted because it was part of a migration:

![Screenshot 2024-05-30 at 3 57 46 PM](https://github.com/SU-HSDO/suhumsci/assets/107938318/eb5487d4-6a97-400e-9cc8-f10fdd00395f)

8. Verify that the "Remove" button is not available next to the media item name:

![Screenshot 2024-05-30 at 3 58 26 PM](https://github.com/SU-HSDO/suhumsci/assets/107938318/e973fbcd-4451-490d-9622-34ba70bf05ce)

9. Verify that other fields for the image can be edited, such as Focal Point, alt text, and caption.
10. Optionally, log in to the site as an admin and verify that admins can delete media from the contextual menu and from the media edit form.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207858685626376
